lib/fastlane/plugin/fueled/helper/fueled_helper.rb

```diff
@@ -155,6 +155,190 @@ def self.default_gh_release_output_files
         ].compact!
       end
 
+      # Calculate the code coverage percentage
+      # This function takes the code coverage configuration file and checks the result bundle file for the code coverage percentage using the xccov tool.
+      # Additionally, this function can also generate HTML reports for the code coverage.
+      def self.calculate_code_coverage_percentage(
+        code_coverage_config_file_path,
+        result_bundle_file_path,
+        minimum_code_coverage_percentage = nil,
+        generate_reports = false
+      )
+        if !File.exist? code_coverage_config_file_path
+          UI.user_error!("Cannot find Fueled code coverage config file at #{code_coverage_config_file_path}")
+        end
+
+        if !File.exist? result_bundle_file_path
+          UI.user_error!("Cannot find result bundle file at #{result_bundle_file_path}")
+        end
+
+        code_coverage_config_file = File.read(code_coverage_config_file_path)
+
+        if !code_coverage_config = JSON.parse(code_coverage_config_file) rescue nil
+            UI.user_error!('The provided Fueled code coverage config file isn\'t in a valid format')
+        end 
+
+        UI.message("Calculating Code Coverage Percentage... 🤓")
+        test_results_json_string = `xcrun xccov view --report --json \"#{result_bundle_file_path}\"`
+        test_results = JSON.parse(test_results_json_string)
+
```

Comment:
This seems like a shell command, although it's being parsed. I got a little confused here. @hadiidbouk Can you explain this for me?

Reply:
@sinchan-fueled Sure! 

You can see that we add  `--json`  to the command, which will convert the output of the command into a JSON format; we are using `JSON.parse` here to convert the JSON string into a JSON object.

We need the JSON output of the `xccov` command for easy access to the data; here, we want to see what files have been covered when running our tests.

Reply:
Ohh, I forgot to mention a backtick will execute the command, and it returns the output :)

Reply:
Gotchu! The backtick comment cleared it for me! Thanks Hadi 😄